File: lab4/zadanie1/python/main.py
from prime_numbers import (
		PrimeNumbers, 
		CheckParams,
		Prime,
		IsPrime
		)
import sys

def main():
	if not CheckParams(sys.argv):
		print("Niepoprawne parametry!")
		return -1
	n = int(sys.argv[2])

	if sys.argv[1] == "pn":
		print(PrimeNumbers(n))
	elif sys.argv[1] == "pr":
		print(Prime(n))
	else:
		print(IsPrime(n))

if __name__ == "__main__":
	main()

File: lab4/zadanie1/c/main.c
#include <stdio.h>
#include <string.h>

#include "primenumbers.h"

int main(int argc, char *argv[]){
  unsigned n = 0;
  if (argc != 3){
    printf("Zła liczba parametrów\n");
    return -1;
  }
  if (!valid_number(argv[2])){
    return -1;
  }
  sscanf(argv[2], "%u", &n);
  if (strcmp(argv[1], "pn") == 0){
    printf("%u", PrimeNumbers(n));
  }
  else if (strcmp(argv[1], "pr") == 0){
    if (n!=0){
      printf("%u", Prime(n));
    }
    else {
      printf("Podaj numer liczby pierwszej!\n");
    }
  }
  else if (strcmp(argv[1], "ip") == 0){
    if (IsPrime(n)){
      printf("true\n");
    }
    else {
      printf("false\n");
    }
  }
  else {
    printf("Błędny parametr");
    return -1;
  }
  return 0;
}
